src/test/regress/expected/failure_1pc_copy_hash.out

```diff
@@ -0,0 +1,384 @@
+ count 
+-------
+     4
+(1 row)
+
+SELECT citus.mitmproxy('conn.onQuery(query="^COMMIT$").killall()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+COPY copy_test FROM PROGRAM 'echo 0, 0 && echo 1, 1 && echo 2, 4 && echo 3, 9' WITH CSV;
+WARNING:  connection not open
+CONTEXT:  while executing command on localhost:57640
+WARNING:  failed to commit critical transaction on localhost:57640, metadata is likely out of sync
+WARNING:  connection not open
+CONTEXT:  while executing command on localhost:57640
+-- the shard is marked invalid
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT count(1) FROM pg_dist_shard_placement WHERE shardid IN (
+  SELECT shardid FROM pg_dist_shard WHERE logicalrelid = 'copy_test'::regclass
+) AND shardstate = 3;
+ count 
+-------
+     1
+(1 row)
+
+SELECT count(1) FROM copy_test;
+ count 
+-------
+     8
+(1 row)
+
+-- ==== clean up a little bit before running the next test ====
+UPDATE pg_dist_shard_placement SET shardstate = 1
+WHERE shardid IN (
+  SELECT shardid FROM pg_dist_shard WHERE logicalrelid = 'copy_test'::regclass
+);
+TRUNCATE copy_test;
+-- ==== try to COPY invalid data ====
+-- here the coordinator actually sends the data, but then unexpectedly closes the
+-- connection when it notices the data stream is broken. Crucially, it closes the
+-- connection before sending COMMIT, so no data makes it into the worker.
+COPY copy_test FROM PROGRAM 'echo 0, 0 && echo 1, 1 && echo 2, 4 && echo 3, 9 && echo 10' WITH CSV;
+ERROR:  missing data for column "value"
+CONTEXT:  COPY copy_test, line 5: "10"
+-- kill the connection if the coordinator sends COMMIT. It doesn't, so nothing changes
+SELECT citus.mitmproxy('conn.onQuery(query="^COMMIT$").kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+COPY copy_test FROM PROGRAM 'echo 0, 0 && echo 1, 1 && echo 2, 4 && echo 3, 9 && echo 10' WITH CSV;
+ERROR:  missing data for column "value"
+CONTEXT:  COPY copy_test, line 5: "10"
+SELECT * FROM copy_test;
+ key | value 
+-----+-------
+(0 rows)
+
+-- ==== clean up some more to prepare for tests with only one replica ====
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+TRUNCATE copy_test;
+UPDATE pg_dist_shard_placement SET shardstate = 3 WHERE nodeport = :worker_1_port;
+SELECT * FROM pg_dist_shard_placement WHERE shardid IN (
+  SELECT shardid FROM pg_dist_shard WHERE logicalrelid = 'copy_test'::regclass
+) ORDER BY shardid, placementid;
+ shardid | shardstate | shardlength | nodename  | nodeport | placementid 
+---------+------------+-------------+-----------+----------+-------------
+  100400 |          3 |           0 | localhost |    57637 |         100
+  100400 |          1 |           0 | localhost |    57640 |         101
+(2 rows)
+
+-- ==== okay, run some tests where there's only one active shard ====
+COPY copy_test FROM PROGRAM 'echo 0, 0 && echo 1, 1 && echo 2, 4 && echo 3, 9' WITH CSV;
+SELECT * FROM copy_test;
```

Comment:
Minor: Shall we add `ORDER BY`? Though the data comes from a single shard, there is a possibility of ending up with a different order.

Reply:
added!